src/main/kotlin/bsu/cc/parser/XlsxParser.kt

```diff
@@ -0,0 +1,152 @@
+package bsu.cc.parser
+
+import org.apache.poi.ss.usermodel.Cell
+import org.apache.poi.ss.usermodel.CellType
+import org.apache.poi.ss.usermodel.Row
+import org.apache.poi.ss.usermodel.Sheet
+import org.apache.poi.xssf.usermodel.XSSFWorkbook
+import java.io.File
+import java.io.FileInputStream
+import java.lang.IllegalArgumentException
+import java.lang.IllegalStateException
+import java.lang.reflect.Type
+import java.util.*
+
+
+
+fun readWorkbook(fileName: String): XSSFWorkbook {
+    val file = FileInputStream(File(fileName))
+    return XSSFWorkbook(file)
+}
+
+fun <T> sheetToDataClasses(
+        sheet: Sheet,
+        dataProducer: (namedRowMap: Map<String, Cell>) -> T,
+        ignoreDuplicateHeaders: Boolean = false
+): Sequence<T> {
+    val headerMap = sheetHeaderMap(sheet)
+    fun translatedProducer(indexedRowMap: Map<Int, Cell>): T {
```

Comment:
I like this solution. I wonder how kotlin compiles these functions. If they get turned into static method and they just enforce scoping or if they just turn it into a lambda because of the closure?